ctk/common/src/test/kotlin/org/codice/compilance/verification/profile/SingleSignOnProfileVerifierSpec.kt

```diff
@@ -13,161 +13,225 @@
 
-        "response with empty and not empty holder-of-key subject confirmation should fail" {
-            val multipleSubjConf = """
-                |  <s2:SubjectConfirmationData $correctType>
-                |    <ds:KeyInfo>
-                |      <ds:KeyName>key</ds:KeyName>
-                |    </ds:KeyInfo>
-                |  </s2:SubjectConfirmationData>
-                |</s2:SubjectConfirmation>
-                |<s2:SubjectConfirmation Method="$HOLDER_OF_KEY_URI">
-                """.trimMargin()
-
-            Common.buildDom(createResponse(multipleSubjConf)).let {
+        "response with incorrect assertion issuer value should fail" {
+            Common.buildDom(createResponse(false, assertionIssuer = "wrong")).let {
                 shouldThrow<SAMLComplianceException> {
-                    SingleSignOnProfileVerifier(it).verifyHolderOfKey()
-                }.message?.shouldContain(SAMLProfiles_3_1_a.message)
+                    SingleSignOnProfileVerifier(it).verify()
+                }.message?.shouldContain(SAMLProfiles_4_1_4_2_b.message)
             }
         }
 
-        "response with empty and not empty key info subject confirmation data should fail" {
-            val multipleSubjConf = """
-                |  <s2:SubjectConfirmationData $correctType>
-                |    <ds:KeyInfo>
-                |      <ds:KeyName>key</ds:KeyName>
-                |    </ds:KeyInfo>
-                |  </s2:SubjectConfirmationData>
-                |</s2:SubjectConfirmation>
-                |<s2:SubjectConfirmation Method="$HOLDER_OF_KEY_URI">
-                |  <s2:SubjectConfirmationData $correctType/>
-                """.trimMargin()
-
-            Common.buildDom(createResponse(multipleSubjConf)).let {
+        "response with no assertion should fail" {
+            val noAssertionResponse = "<s:Response $responseParams/>"
+            Common.buildDom(noAssertionResponse).let {
                 shouldThrow<SAMLComplianceException> {
-                    SingleSignOnProfileVerifier(it).verifyHolderOfKey()
-                }.message?.shouldContain(SAMLProfiles_3_1_a.message)
+                    SingleSignOnProfileVerifier(it).verify()
+                }.message?.shouldContain(SAMLProfiles_4_1_4_2_d.message)
+            }
+        }
+
+        "unsigned response with encrypted assertion and correct issuer should pass" {
+            Common.buildDom(createResponseWithEncryptedAssertion(correctIdpIssuer)).let {
+                CoreVerifierTest(it).verify()
+                SingleSignOnProfileVerifier(it).verify()
+            }
+        }
+
+        "unsigned response with encrypted assertion and incorrect issuer should fail" {
+            Common.buildDom(createResponseWithEncryptedAssertion(incorrectIdpIssuer)).let {
+                shouldThrow<SAMLComplianceException> {
+                    SingleSignOnProfileVerifier(it).verify()
+                }.message?.shouldContain(SAMLProfiles_4_1_4_2_b.message)
```

Comment:
✏️ Nit: In following the flow of the `SingleSignOnProfileVerifier`, these two new tests should be added above the assertion tests.

Reply:
I decided to put them on the bottom cause they call `CoreVerifier`. I'll comment that